pyro/distributions/delta.py

```diff
@@ -55,7 +55,7 @@ def rsample(self, sample_shape=torch.Size()):
 
     def log_prob(self, x):
         v = self.v.expand(self.shape())
-        log_prob = x.new_tensor(x == v).log()
+        log_prob = (x == v).type(x.dtype).log().to(x.device)
```

Comment:
I'm surprised the `.to(x.device)` is needed.

Reply:
`.to` is only useful for the multi gpu case, and can be avoided if x and v are on the same device (which I suppose they should be unless we are doing something funny). I think we can remove it.